src/encoded/static/components/search.js

```diff
@@ -371,12 +379,14 @@ const ExperimentComponent = (props, reactContext) => {
 ExperimentComponent.propTypes = {
     context: PropTypes.object.isRequired, // Experiment search results
     cartControls: PropTypes.bool, // True if displayed in active cart
+    mode: PropTypes.string, // Special search-result modes, e.g. "picker"
     auditIndicators: PropTypes.func.isRequired, // Audit decorator function
     auditDetail: PropTypes.func.isRequired,
 };
 
 ExperimentComponent.defaultProps = {
     cartControls: false,
+    mode: '',
```

Comment:
I’m not sure what you mean. "mode" now gets passed through to the type-specific search result rendering so that they can check for modes just like the existing top-level facet code has always done.

Reply:
It just didn't seem to me like the "picker" mode was used for anything in this component, so maybe the only thing that was relevant was whether the mode was "cart-view". I could be wrong about this though.

Reply:
No that’s right. "picker" mode is only used at the level covering all search results so that they display differently in the object-editor form. But that’s where `mode` stopped so I now pass `mode` all the way down to the individual search result level, and that’s where 'cart-view' mode is used. So the edit forms can affect the entire search results using mode "picker," and the cart view can affect individual search results using mode "cart-view." The individual results don’t care about "picker" mode. I’m just overloading this `mode` property since their functionalities currently don’t overlap yet they have a similar role.

If we ever have entire search results displayed differently for cart view, or individual search results care about picker mode, then this mechanism won’t work anymore.

Reply:
Ok, this all makes sense. I guess I was suggesting that you just pass a flag saying if the mode was "cart-view" since you don't care if the mode is "picker" for this component. I think also I found the comment `mode: PropTypes.string, // Special search-result modes, e.g. "picker"` a little confusing because the example is not relevant for the component. This isn't very important though, just explaining why I mentioned it.

Reply:
What might be nice is a `mode` _object_, so you can have multiple flags at the same time. That crossed my mind, as did a new property for cart view, but I felt like mode could be more useful than just for one thing, but I didn’t know if making an object was worth how little we use it. Something to think about if we need to expand on this someday.